Customizing/global/plugins/Services/Cron/CronHook/DiMAkImport/classes/Import/Data/ilDB.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace CaT\Plugins\DiMAkImport\Import\Data;
+
+class ilDB implements DB {
+	const TABLE_NAME = "dimak_agent_number";
+
+	public function __construct($db)
+	{
+		$this->db = $db;
+	}
+
+	public function truncate()
+	{
+		$query = "DELETE FROM ".self::TABLE_NAME;
+		$this->db->manipulate($query);
+	}
+
+	public function save($file_path)
+	{
+		$query = "LOAD DATA INFILE ".$this->db->quote($file_path, "text")." INTO TABLE ".self::TABLE_NAME;
```

Comment:
Hmm, das sieht gefährlich aus. Wieviele Zeilen erwartest du da? Wie sieht es mit den einstllngen zu quotes und zeilenenden aus?

Reply:
Laut Konzept kommen hier die Nummern jeweils in einer Zeile zurück.
Bsp.:

32323
2344
23432
234234

Ich hab das bei mir mit 23074 Einträgen probiert. Es war unfassbar schneller als der Import Zeile für Zeile. Umbrüche wurden nicht mit importiert.

Reply:
https://dev.mysql.com/doc/refman/8.0/en/load-data.html

Reply:
Ja, die habe ich gesehen. Geschwindigkeit ist, glaube ich, egal, da es eh nur nächtens ein mal läuft. Es ist halt gefährlich, weil wir nicht wissen wie sich das auf verschiedenen systemen verhält (linux vs unix vs windows),

Reply:
Zitat: Es wird davon ausgegangen, dass eine Vermittlernummer nicht mehr als 32 Zeichen enthält. Werden hier längere Vermittlernummern übertragen oder enthält die Datei eine Vermittlernummer, die nicht ASCII-kodiert ist, wird ein entsprechender Logeintrag geschrieben. Die entsprechende Zeile wird dann nicht in die Datenbank der GOA übertragen, ohne dass die gesamte Verarbeitung abbricht.
Hab ich das Übersehen?

Reply:
Ich habs geändert. Den Part hatte ich übersehen.